setupext.py

```diff
@@ -101,6 +100,31 @@ def download_or_cache(url, sha):
     return BytesIO(data)
 
 
+def get_and_extract_tarball(urls, sha, dirname):
+    toplevel = Path("build", dirname)
```

Comment:
Even though this is not user-facing API, it would be good to have a docstring describing what this does. - In particular also because the meaning of multiple urls is not obvious.

Reply:
sure